dear-diary-app-ts/src/components/SigninCard.tsx

```diff
@@ -0,0 +1,101 @@
+import React, { useState,ReactElement } from 'react';
+import Box from '@mui/material/Box';
+import Paper from '@mui/material/Paper';
+import { styled } from '@mui/material/styles';
+import TextField from '@mui/material/TextField';
+import Button from '@mui/material/Button';
+import ganarateName from "../util/ganarateName";
+import { Link } from 'react-router-dom';
+
+
+
+// styles loging card components
+const Item = styled(Paper)(({ theme }) => ({
+    ...theme.typography.body2,
+    textAlign: 'center',
+    color: theme.palette.text.secondary,
+    height: 60,
+    lineHeight: '60px',
+}));
+  
+
+
+interface SigninCardProps {
+  nameHandler: (name: string) => void;
+}
+
+function SigninCard({ nameHandler }:SigninCardProps): ReactElement {
+
+
+    // these are states for our local value
+    const [genarateRandomName, setGenarateRandomName] = useState('');
+    // console.log(genarateName);
+
+
+
+//set to ganerated name to usestate
+function g_click() { 
```

Comment:
all functions need to be camelCase, I'll merge this for now, and fix this in the next PR